services/src/test/java/se/inera/intyg/common/services/messages/DefaultCertificateMessagesProviderTest.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2022 Inera AB (http://www.inera.se)
+ *
+ * This file is part of sklintyg (https://github.com/sklintyg).
+ *
+ * sklintyg is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * sklintyg is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package se.inera.intyg.common.services.messages;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+
+class DefaultCertificateMessagesProviderTest {
+
+    @Test
+    void shallReplaceDynamicKeyWithValue() {
```

Comment:
Good with a test class as it seemed to be missing! Maybe you could introduce a few more test to cover the different scenarios. The Provider is very testable so it is quite easy to create more scenarios.